lib/gi/client.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+require 'common/client/base'
+require 'gi/configuration'
+
+module Gi
+  # Core class responsible for api interface operations
+  class Client < Common::Client::Base
+    configuration Gi::Configuration
+
+    def get_autocomplete_suggestions(term)
+      json_hash = perform(:get, 'institutions/autocomplete', {term: term}, nil).body
```

Comment:
What happens if this request errors out, like 404 or 500?  `GiController` looks like it will just `render json:` blindly so it will be a `200` response with some error json.